packages/openapi-v3-types/test/unit/type-guards.unit.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/openapi-v3-types
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  SchemaObject,
+  isSchemaObject,
+  ReferenceObject,
+  isReferenceObject,
+} from '../../src';
+
+describe('type-guards unit tests', () => {
+  describe('isSchemaObject()', () => {
+    it('returns true for a schema object', () => {
+      const schemaObject = new TestSchemaObject();
+      expect(isSchemaObject(schemaObject)).to.be.True();
+    });
+
+    it('returns false for a reference object', () => {
+      const referenceObject = new TestReferenceObject();
+      expect(isSchemaObject(referenceObject)).to.be.False();
+    });
+  });
+
+  describe('isReferenceObject()', () => {
+    it('returns true for a reference object', () => {
+      const referenceObject = new TestReferenceObject();
+      expect(isReferenceObject(referenceObject)).to.be.True();
+    });
+
+    it('returns false for a schema object', () => {
+      const schemaObject = new TestSchemaObject();
+      expect(isReferenceObject(schemaObject)).to.be.False();
+    });
+  });
+
+  class TestSchemaObject implements SchemaObject {}
```

Comment:
I don't know too much about `SchemaObject`, but shouldn't the test here be more than just an empty class?

Reply:
Nope, `SchemaObject` doesn't have any required properties